src/spanning_trees.c

```diff
@@ -366,3 +369,137 @@ int igraph_i_minimum_spanning_tree_prim(const igraph_t* graph,
 }
 
 
+/* igraph_random_spanning_tree */
+
+/* Loop-erased random walk (LERW) implementation.
+ * res must be an initialized vector. The edge IDs of the spanning tree
+ * will be added to it. res will not be cleared before doing this. */
+static int igraph_i_lerw(const igraph_t *graph, igraph_vector_t *res, igraph_integer_t start,
+                  igraph_integer_t comp_size, igraph_vector_bool_t *visited, const igraph_inclist_t *il)
+{
+    igraph_integer_t visited_count;
+
+    IGRAPH_CHECK(igraph_vector_reserve(res, igraph_vector_size(res) + comp_size-1));
+
+    RNG_BEGIN();
+
+    VECTOR(*visited)[start] = 1;
+    visited_count = 1;
+
+    while (visited_count < comp_size) {
+        long degree, edge, idx;
+        igraph_integer_t next;
+        igraph_vector_int_t *edges;
+
+        edges = igraph_inclist_get(il, start);
+
+        /* choose a random edge */
+        degree = igraph_vector_int_size(edges);
+        idx = RNG_INTEGER(0, degree-1);
+        edge = VECTOR(*edges)[idx];
+
+        /* set 'start' to the next vertex */
+        next = IGRAPH_TO(graph, edge);
```

Comment:
We have an `IGRAPH_OTHER(graph, edge, start)` macro that gives you the _other_ endpoint of an edge.